# HW10

### Jingheng Huan
NetID: jh730

Q1: Derive a standard form LP model with two nonnegative decision
variables, {x1 ≤ 12, x2 ≤ 16}, for maximizing the following sum: 36x1 + 25x2. A
solution is feasible if the sum of the decision variables is at most 20.

Maximize: Z = 36x<sub>1</sub> + 25x<sub>2</sub>

Subject to the constraints:

x<sub>1</sub> + x<sub>2</sub> ≤ 20 (The sum of the decision variables is at most 20)

x<sub>1</sub> ≤ 12

x<sub>2</sub> ≤ 16

x<sub>1</sub>, x<sub>2</sub> ≥ 0 (Non-negativity constraints)

This is the standard form Linear Programming model for the given problem. The objective is to find the values of x<sub>1</sub> and x<sub>2</sub> that maximize Z, subject to the constraints.

Q2: Perform the Simplex algorithm to solve the corresponding
LP model from the previous question.

In [2]:
from scipy.optimize import linprog

# Coefficients of the objective function
c = [-36, -25] # We use negative Coefficients because linprog minimizes the objective function

# Coefficients of the inequality constraints
A = [[1, 1], [1, 0], [0, 1]]

# Right hand side of the inequality constraints
b = [20, 12, 16]

# Bounds for the variables
x0_bounds = (0, None)
x1_bounds = (0, None)

res = linprog(c, A_ub = A, b_ub = b, bounds = [x0_bounds, x1_bounds], method = 'highs')

print('Optimal value:', -res.fun, '\nx values:', res.x)

Optimal value: 632.0 
x values: [12.  8.]


The optimal solution to the linear programming problem is when x<sub>1</sub> = 12, x<sub>2</sub> = 8, and Z = 36(12) + 25(8) = 632.